Add HCP mode documentation (`v1.2.0`) (#178)
rancher/k3k-product-docs: versions/v1.2.0/modules/en/pages/how-tos-for-admin/choose-mode.adoc

_"I'm testing a new admission controller and policy engine before rolling it out cluster-wide."_ +
→ Use *Virtual mode*, if you need to test cluster-wide policies, custom admission flow, or advanced extensions with full control.

== HCP Mode

> NOTE: HCP mode is *experimental* in {product-name} v1.2.0.

Reviewer: ```suggestion
[NOTE]
====
HCP mode is *experimental* in {product-name} v1.2.0.
====
```

*Best for:*

* Users who want K3k to manage the control plane lifecycle but need *real, external* worker nodes
* Edge and IoT scenarios where workers run outside the host data center
* Evaluating workloads on dedicated bare-metal or VM hardware while keeping the apiserver centrally managed

> In *HCP mode*, K3k runs an agentless K3s control plane inside the host cluster, and you supply the worker nodes. External workers join the cluster using the standard K3s installer script. The apiserver must be reachable from your worker network via `nodePort`, `loadBalancer`, or `ingress`. More details on the link:./../architecture.adoc#_hcp_mode[architecture] and the step-by-step xref:how-tos-for-user/create-hcp-virtual-cluster.adoc[Creating an HCP Virtual Cluster] guide.

Reviewer: ```suggestion
In *HCP mode*, K3k runs an agentless K3s control plane inside the host cluster, and you supply the worker nodes. External workers join the cluster using the standard K3s installer script. The apiserver must be reachable from your worker network via `nodePort`, `loadBalancer`, or `ingress`. For more information, see xref:./../architecture.adoc#_hcp_mode[architecture] and the step-by-step xref:how-tos-for-user/create-hcp-virtual-cluster.adoc[Creating an HCP Virtual Cluster] guide.
```

_"I'm testing a new admission controller and policy engine before rolling it out cluster-wide."_ +
→ Use *Virtual mode*, if you need to test cluster-wide policies, custom admission flow, or advanced extensions with full control.

== HCP Mode

[NOTE]
====
HCP mode is *experimental* in {product-name} v1.2.0.
====

*Best for:*

* Users who want K3k to manage the control plane lifecycle but need *real, external* worker nodes
* Edge and IoT scenarios where workers run outside the host data center
* Evaluating workloads on dedicated bare-metal or VM hardware while keeping the apiserver centrally managed

> In *HCP mode*, K3k runs an agentless K3s control plane inside the host cluster, and you supply the worker nodes. External workers join the cluster using the standard K3s installer script. The apiserver must be reachable from your worker network via `nodePort`, `loadBalancer`, or `ingress`. For more information, see xref:./../architecture.adoc#_hcp_mode[architecture] and the step-by-step xref:how-tos-for-user/create-hcp-virtual-cluster.adoc[Creating an HCP Virtual Cluster] guide.

Reviewer 2: Should be fine. It's at the version root.
```suggestion
In *HCP mode*, K3k runs an agentless K3s control plane inside the host cluster, and you supply the worker nodes. External workers join the cluster using the standard K3s installer script. The apiserver must be reachable from your worker network via `nodePort`, `loadBalancer`, or `ingress`. For more information, see xref:/architecture.adoc#_hcp_mode[architecture] and the step-by-step xref:how-tos-for-user/create-hcp-virtual-cluster.adoc[Creating an HCP Virtual Cluster] guide.
```
